### ISSUE TITLE
FATAL ERROR: CALL_AND_RETRY_LAST Allocation failed - JavaScript heap out of memory

### ISSUE THREAD

**Issue by luisfuertes** [nozzle/react-static#703]
Hi, when i try to do `yarn run build` have the next error:

```
=> Copying public directory...
=> [✓] Public directory copied: 2.035ms
=> Building Routes...
=> [✓] Routes Built: 224.641ms
=> Bundling App...
=> [✓] App Bundled: 78788.254ms
=> Fetching Site Data...
=> [✓] Site Data Downloaded: 0.173ms
=> Fetching Route Data...
=> [=======] 7/7 100% 7000/s 0.0s 
=> [✓] Route Data Downloaded: 5.108ms
=> Exporting Route Data...
=> [✓] Route Data Exported: 1.069ms

<--- Last few GCs --->

[5656:0x102801e00]   347957 ms: Mark-sweep 1315.5 (1430.8) -> 1315.4 (1430.8) MB, 1772.7 / 0.0 ms  allocation failure GC in old space requested
[5656:0x102801e00]   349929 ms: Mark-sweep 1315.4 (1430.8) -> 1315.4 (1424.3) MB, 1971.4 / 0.0 ms  last resort GC in old space requested
[5656:0x102801e00]   351926 ms: Mark-sweep 1315.4 (1424.3) -> 1315.4 (1419.8) MB, 1996.9 / 0.0 ms  last resort GC in old space requested


<--- JS stacktrace --->

==== JS stack trace =========================================

Security context: 0x178f483257c1 <JSObject>
    1: _append [/Users/mobile3/Desarrollo/Folder/proyect/node_modules/babel-generator/lib/buffer.js:~94] [pc=0x34b82f0da13](this=0x178f17b23041 <Buffer map = 0x178fb3745281>,str=0x178f4836a769 <String[1]:  >,line=130105,column=0,identifierName=0x178f94502201 <null>,filename=0x178f94502201 <null>)
    2: Identifier [/Users/mobile3/Desarrollo/Folder/proyect/node_modules/babel-generat...

FATAL ERROR: CALL_AND_RETRY_LAST Allocation failed - JavaScript heap out of memory
 1: node::Abort() [/usr/local/bin/node]
 2: node::FatalException(v8::Isolate*, v8::Local<v8::Value>, v8::Local<v8::Message>) [/usr/local/bin/node]
 3: v8::internal::V8::FatalProcessOutOfMemory(char const*, bool) [/usr/local/bin/node]
 4: v8::internal::Factory::NewUninitializedFixedArray(int) [/usr/local/bin/node]
 5: v8::internal::(anonymous namespace)::ElementsAccessorBase<v8::internal::(anonymous namespace)::FastPackedObjectElementsAccessor, v8::internal::(anonymous namespace)::ElementsKindTraits<(v8::internal::ElementsKind)2> >::GrowCapacity(v8::internal::Handle<v8::internal::JSObject>, unsigned int) [/usr/local/bin/node]
 6: v8::internal::Runtime_GrowArrayElements(int, v8::internal::Object**, v8::internal::Isolate*) [/usr/local/bin/node]
 7: 0x34b824042fd
 8: 0x34b82f0da13
 9: 0x34b82f3d7b5
10: 0x34b8240535f
✨  Done in 357.19s.

```

I have tried to add bluebird Promise.map, return empty data and only one static route but crash continue.

### Environment
1.  `react-static -V: 5.9.7`
2.  `node -v`: 8.11.1
3.  `yarn --version or npm -v: 1.6.0`
4.  `Operating system: MacOS High Sierra v10.13.5`

I cant share my repository but if need more info ask me.

Any ideas about it? Thanks

**Comment by luisfuertes** [comment]
How can i add `node --max-old-space-size=1024 my-node-script.js` #increase to 1gb for react-static?

**Comment by luisfuertes** [comment]
Ok i have tried `node --max-old-space-size=2048 /Users/mobile3/.config/yarn/global/node_modules/react-static/bin/react-static-build` but error continue

**Comment by bkniffler** [comment]
I've been having similar errors when in dev mode and modifying static.config.js. There must be memory leaking since some beta version.

**Comment by tannerlinsley** [comment]
I'll be pushing another beta release soon. Stay tuned.

**Comment by luisfuertes** [comment]
Finally i add `export NODE_OPTIONS=--max_old_space_size=4096` to my `.bash_profile` file

**Comment by tannerlinsley** [comment]
This shouldn't be a problem any more, but if you are building a massive site, you may need to do as suggested above and increase your memory max sizes. This is common across all JS generators unfortunately. But I promise RS is as performant as it can be at the moment :)

**Comment by simonunlikely** [comment]
Hey,

I'm experiencing this issue on a react-static 7.0.9 projet while trying to export 50k pages with the same template and same data (was doing a stress test to get an idea of the build time).
Runs fine with 10k though. Noticed if I set the maxThreads option to 1 I don't get the error but it is much slower.

Is there anything I can do ? I've already tried to set the node max memory to something bigger with no success.

Thanks !
